zanata-war/src/main/java/org/zanata/adapter/PropertiesAdapter.java

```diff
@@ -0,0 +1,225 @@
+        } catch (IOException e) {
+            throw new FileFormatAdapterException(
+                "Could not open the URL. The URL is OK but the input stream could not be opened.\n"
+                    + e.getMessage(), e);
+        } finally {
+            try {
+                inputStream.close();
+            } catch (IOException e1) {
+            }
+        }
+        return doc;
+    }
+
+    public TranslationsResource parseTranslationFile(URI fileUri,
+        String localeId, Optional<String> params, String charset)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        if (StringUtils.isEmpty(localeId)) {
+            throw new IllegalArgumentException(
+                "locale id string cannot be null or empty");
+        }
+
+        PropReader propReader = new PropReader(charset, LocaleId.EN_US,
+            ContentState.Approved);
+
+        BufferedInputStream inputStream = null;
+
+        /**
+         * Resource is not needed for properties file translation parser
+         * as its only used for contentHash check
+         */
+        Resource srcDoc = new Resource();
+
+        TranslationsResource targetDoc = new TranslationsResource();
+
+        try {
+            inputStream = readStream(fileUri);
+            propReader.extractTarget(targetDoc, inputStream, srcDoc);
+        } catch (IOException e) {
+            throw new FileFormatAdapterException(
+                "Could not open the URL. The URL is OK but the input stream could not be opened.\n"
+                    + e.getMessage(), e);
+        } finally {
+            try {
+                inputStream.close();
+            } catch (IOException e1) {
+            }
+        }
+        return targetDoc;
+    }
+
+    public void writeTranslatedFile(OutputStream output, Resource resource,
+        TranslationsResource translationsResource, String locale,
+        Optional<String> params, String charset)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        //write source string with empty translation
+        boolean createSkeletons = true;
+
+        File tempFile = null;
+        try {
+            tempFile = File.createTempFile("filename", "extension");
+            if(charset.equals(ISO_8859_1)) {
+                PropWriter.writeTranslationsFile(resource,
+                        translationsResource, tempFile,
+                        PropWriter.CHARSET.Latin1, createSkeletons);
+            } else if (charset.equals(UTF_8)) {
+                PropWriter.writeTranslationsFile(resource,
+                    translationsResource,
+                    tempFile, PropWriter.CHARSET.UTF8, createSkeletons);
+            }
+
+            byte[] buffer = new byte[4096]; // To hold file contents
+            int bytesRead;
+            FileInputStream input = new FileInputStream(tempFile);
+            while ((bytesRead = input.read(buffer)) != -1) {
+                output.write(buffer, 0, bytesRead);
```

Comment:
- [ ] This logic is used in a few places, and is too low-level to put with this code anyway. Better extract it to a utility method `writeFileToOutputStream(File file, OutputStream output)`.
- [ ] Was this copied from one of the existing adapters? If so, might be good to replace that with a method call too.